mesonbuild/scripts/meson_exe.py

```diff
@@ -64,14 +65,30 @@ def run_exe(exe):
         child_env['PATH'] = (os.pathsep.join(exe.extra_paths + ['']) +
                              child_env['PATH'])
         if exe.exe_runner and mesonlib.substring_is_in_list('wine', exe.exe_runner.get_command()):
-            wine_paths = ['Z:' + p for p in exe.extra_paths]
-            wine_path = ';'.join(wine_paths)
-            # Don't accidentally end with an `;` because that will add the
-            # current directory and might cause unexpected behaviour
-            if 'WINEPATH' in child_env:
-                child_env['WINEPATH'] = wine_path + ';' + child_env['WINEPATH']
-            else:
-                child_env['WINEPATH'] = wine_path
+            seen = set()
+            wine_paths = ['Z:' + p for p in exe.extra_paths if not (p in seen or seen.add(p))]
+            wine_paths += [p for p in child_env.get('WINEPATH', '').split(';') if not (p in seen or seen.add(p))]
+
+            getShortPathScript = '%s.bat' % str(uuid.uuid4()).lower()[:5]
```

Comment:
We don't need to run a batch script to get short paths. I implemented the same thing for `PATH` in gst-uninstalled: https://gitlab.freedesktop.org/gstreamer/gst-build/blob/master/scripts/common.py#L13

Reply:
We are running on linux here... we can not do what you did there.